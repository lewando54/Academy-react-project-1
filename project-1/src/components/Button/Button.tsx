import React, {useState, useEffect} from 'react';
import styled from 'styled-components'

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    children: string,
    color: string,
}

const StyledButton = styled.button`
    color: white;
    border-radius: 0.375rem;
    width: 100%;
    text-transform: uppercase;
    font-size: 0.875rem;
    line-height: 1.25rem;
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    box-shadow: 0 4px 6px rgba(0,0,0,0.3);
    letter-spacing: 0.025em;
`;

function Button({color, children, ...props}: ButtonProps){
    const [bgColor, setBgColor] = useState('');
    const colorOptions: any = {
        'primary': 'rgb(236 72 153)',
        'secondary': 'rgb(229 231 235)'
    }
    useEffect(() => {
        setBgColor(color);
    }, [color])

    return (
        <StyledButton {...props} style={{backgroundColor: colorOptions[bgColor]}}>
            {children}
        </StyledButton>
    )
}

export default Button