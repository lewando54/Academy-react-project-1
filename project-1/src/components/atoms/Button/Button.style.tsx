import styled from 'styled-components'

export const StyledButton = styled.button`
    color: white;
    border-radius: 0.375rem;
    width: 100%;
    text-transform: uppercase;
    font-size: 0.875rem;
    line-height: 1.25rem;
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    box-shadow: 0 4px 6px rgba(0,0,0,0.3);
    letter-spacing: 0.025em;
    border: 0;
    cursor: pointer;
`
