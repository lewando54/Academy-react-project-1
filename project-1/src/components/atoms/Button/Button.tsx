import React from 'react'
import { StyledButton } from './Button.style'

type colorProp = 'primary' | 'secondary'
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: string
  color: colorProp
}

function Button ({ color, children, onClick }: ButtonProps): React.ReactElement {
  const colorOptions: { [key in colorProp]: string } = {
    primary: 'rgb(236 72 153)',
    secondary: 'rgb(229 231 235)'
  }

  return (
        <StyledButton onClick={onClick} style={{ backgroundColor: colorOptions[color] }}>
            {children}
        </StyledButton>
  )
}

export default Button
