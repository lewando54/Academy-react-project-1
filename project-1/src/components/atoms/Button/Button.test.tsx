import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import { render, fireEvent, screen } from '@testing-library/react'
import Button from './Button'

describe('Button', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button testId='testxd' color="primary">Click me</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with primary color by default', () => {
    const tree = renderer.create(<Button testId='testxd'>Click me</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render innerHTML text', () => {
    const { getByTestId } = render(<Button testId='testxd' color="secondary">Click me</Button>)
    expect(getByTestId('testxd')).toHaveTextContent('Click me')
  })

  it('should render with primary color', () => {
    render(<Button testId='button' color={'primary'}>Click me</Button>)
    expect(screen.getByTestId('button')).toHaveStyle({ backgroundColor: 'var(--primary)' })
  })

  it('should render with secondary color', () => {
    render(<Button testId='button' color={'secondary'}>Click me</Button>)
    expect(screen.getByTestId('button')).toHaveStyle({ backgroundColor: 'var(--secondary)' })
  })

  it('should call onClick function when clicked', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(<Button testId='testxd' onClick={onClick}>Click me</Button>)
    fireEvent.click(getByTestId('testxd'))
    expect(onClick).toHaveBeenCalled()
  })

})
