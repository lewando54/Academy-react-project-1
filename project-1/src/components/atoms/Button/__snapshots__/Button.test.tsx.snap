// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-bcXHqe iJjCPT"
  color="primary"
  data-testid="testxd"
>
  Click me
</button>
`;

exports[`Button should render with primary color by default 1`] = `
<button
  className="sc-bcXHqe iJjCPT"
  color="primary"
  data-testid="testxd"
>
  Click me
</button>
`;
